homework-submission/React_HW2/todoApp2.jsx

```diff
@@ -0,0 +1,153 @@
+'use strict';
+
+// ContentDynamicTodo component is for map function to use
+const ContentDynamicTodo = ({ description, deadline, className, deleteItem }) => {
+  return (
+    <tr className={className} onClick={deleteItem}>
+      <td className="content"> {description} </td>
+      <td className="content"> {deadline} </td>
+      <td className="content"> {className} </td>
+    </tr>
+  );
+};
+
+// DynamicList where map function is
+const DynamicList = ({ list, deleteItem }) => {
+  const runTodoList = list.map(todo => (
+    <ContentDynamicTodo
+      key={todo.id}
+      description={todo.description}
+      deadline={todo.deadline}
+      className={todo.done ? '√' : 'x'}
+      deleteItem={() => {
+        deleteItem(todo.id);
+      }}
+    />
+  ));
+  return runTodoList;
+};
+
+// Here is our main Application component as a class
+class AppTodo2 extends React.Component {
+  //TodoList which contains new item as state
+  constructor(props) {
+    super(props);
+    this.state = {
+      todoList: [
+        {
+          id: 1,
+          description: 'Get out of bed',
+          deadline: '2017-09-11',
+          done: true,
+        },
+        {
+          id: 2,
+          description: 'Brush teeth',
+          deadline: '2017-09-10',
+          done: false,
+        },
+        {
+          id: 3,
+          description: 'Eat breakfast',
+          deadline: '2017-09-09',
+          done: false,
+        },
+      ],
+    };
+  }
+
+  newTodo = {
+    id: '',
+    description: 'Description',
+    deadline: 'No Deadline Entry',
+    done: false,
+  };
+
+  NewTodoDescriptionAndDeadline = () => {
+    return (
+      <div>
+        <div>
+          <label> Description </label>
+          <div id="containerDescriptionInput">
+            <input
+              type="text"
+              onChange={event => {
+                const inputValue = event.target.value;
+                this.setState({
+                  newTodo: {
+                    ...this.state.newTodo,
+                    description: inputValue,
+                    id: Math.random(),
+                  },
+                });
+              }}
+            />
+          </div>
+        </div>
+        <div>
+          <label htmlFor=""> Deadline </label>
+          <div id="containerDateInput">
+            <input
+              id="inputDate"
+              type="date"
+              onChange={event => {
+                const inputValue = event.target.value;
+                this.setState({
+                  newTodo: {
+                    ...this.state.newTodo,
+                    deadline: inputValue,
+                  },
+                });
+              }}
+            />
+          </div>
+        </div>
+      </div>
+    );
+  };
+
+  // function for adding todo
+  addState = () => {
```

Comment:
When I press `Add` and I don't have any string to add, but istead I'm getting a blank page with this error in the console:
<img width="736" alt="Screen Shot 2019-07-19 at 07 29 47" src="https://user-images.githubusercontent.com/25272822/61511430-03f64c00-a9f7-11e9-9f64-c47f10f1666f.png">


I think you probably need to initial the `newTodo` with the state.

Reply:
Let me go through on it and get back to you